atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WriteOnceDeleteOnceWorkflows.java

```diff
@@ -0,0 +1,106 @@
+import java.util.function.IntFunction;
+
+/**
+ * The idea of this class is to catch bugs around tombstone drops, by having a
+ * workflow which _mostly_ writes once then deletes once for a given cell. Probability and randomness are used to
+ * enforce this behaviour, thus it's possible to have a write without a delete, or visa versa; some cells may
+ * randomly be written or deleted multiple times as well.
+ */
+public final class WriteOnceDeleteOnceWorkflows {
+
+    private static final SecureRandom RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
+
+    private static final Integer COLUMN = 0;
+
+    private WriteOnceDeleteOnceWorkflows() {
+        // Utility class
+    }
+
+    public static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor) {
+        return create(store, configuration, executionExecutor, DefaultBuggifyFactory.INSTANCE, RANDOM::nextInt);
+    }
+
+    @VisibleForTesting
+    static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor,
+            BuggifyFactory buggifyFactory,
+            IntFunction<Integer> randomNumberGenerator) {
+        WriteOnceDeleteOnceWorkflow workflow =
+                new WriteOnceDeleteOnceWorkflow(buggifyFactory, randomNumberGenerator, configuration);
+        return DefaultWorkflow.create(store, workflow::run, configuration, executionExecutor);
+    }
+
+    private static class WriteOnceDeleteOnceWorkflow {
+        private final BuggifyFactory buggifyFactory;
+        private final IntFunction<Integer> rowKeyGenerator;
+        private final WriteOnceDeleteOnceWorkflowConfiguration configuration;
+
+        public WriteOnceDeleteOnceWorkflow(
+                BuggifyFactory buggifyFactory,
+                IntFunction<Integer> rowKeyGenerator,
+                WriteOnceDeleteOnceWorkflowConfiguration configuration) {
+            this.buggifyFactory = buggifyFactory;
+            this.rowKeyGenerator = rowKeyGenerator;
+            this.configuration = configuration;
+        }
+
+        public Optional<WitnessedTransaction> run(InteractiveTransactionStore store, int taskIndex) {
+            configuration.transactionRateLimiter().acquire();
+            return store.readWrite(txn -> {
+                Integer rowKey = rowKeyGenerator.apply(configuration.maxKey());
+                if (buggifyFactory
+                        .maybe(calculateChanceForDelete(taskIndex, configuration.iterationCount()))
+                        .asBoolean()) {
+                    txn.delete(configuration.tableConfiguration().tableName(), cell(rowKey));
+                } else {
+                    txn.write(configuration.tableConfiguration().tableName(), cell(rowKey), RANDOM.nextInt(10000));
```

Comment:
Using `RANDOM` here as we really never need this to be deterministic for testing purposes. This also makes usages of `rowKeyGenerator` for tests a lot easier to reason about / think.